packages/electron/src/electron/main/index.ts

```diff
@@ -1,12 +1,82 @@
-import fs from "node:fs";
 import path from "node:path";
 import * as Sentry from "@sentry/electron/main";
 import { clearBuffer, setupSidecar } from "@spotlightjs/sidecar";
 import { BrowserWindow, Menu, Tray, app, dialog, ipcMain, nativeImage, shell } from "electron";
 import Store from "electron-store";
+import { autoUpdater } from "electron-updater";
 
 const store = new Store();
 
+autoUpdater.forceDevUpdateConfig = process.env.NODE_ENV === "development";
+
+const ONE_HOUR = 60 * 60 * 1000;
+
+function installAndRestart() {
+  /**
+   * On macOS 15+ auto-update / relaunch issues:
+   * - https://github.com/electron-userland/electron-builder/issues/8795
+   * - https://github.com/electron-userland/electron-builder/issues/8997
+   */
+  if (process.platform === "darwin") {
+    app.removeAllListeners("before-quit");
+    app.removeAllListeners("window-all-closed");
+
+    for (const win of BrowserWindow.getAllWindows()) {
+      if (win.isDestroyed()) continue;
+      win.removeAllListeners("close");
+      win.close();
+    }
+  }
+
+  autoUpdater.quitAndInstall();
+}
+
+async function checkForUpdates() {
+  try {
+    await autoUpdater.checkForUpdates();
+  } catch (error) {
+    console.error(error);
+    Sentry.captureException(error);
+  }
+
+  setTimeout(checkForUpdates, ONE_HOUR);
+}
+
+app.on("ready", () => {
+  checkForUpdates();
+
+  autoUpdater.on("update-downloaded", async () => {
+    const result = await dialog.showMessageBox({
+      type: "question",
+      message: "A new update has been downloaded. It will be installed on restart.",
+      buttons: ["Restart", "Later"],
+    });
+
+    if (result.response === 0) {
+      installAndRestart();
+    }
+  });
+
+  autoUpdater.on("update-downloaded", () => {
+    const menuItem = isMac ? { ...template[0].submenu?.[1] } : undefined;
+
+    if (!menuItem) return;
+
+    menuItem.label = "Restart to Update";
+    menuItem.click = () => {
+      installAndRestart();
+    };
+
+    const _template = [...template];
+    if (_template[0].submenu?.[1].id === "check-for-updates") _template[0].submenu[1] = menuItem;
+    Menu.setApplicationMenu(Menu.buildFromTemplate(_template));
+  });
+
+  autoUpdater.on("error", error => {
+    Sentry.captureException(error);
+  });
```

Comment:
Let's drop the braces, we are no longer teenagers :P

```suggestion
  autoUpdater.on("error", error => Sentry.captureException(error));
```